webapp/publisher/snaps/views.py

```diff
@@ -121,7 +121,8 @@ def publisher_snap_metrics(snap_name):
     metrics_query_json = metrics_helper.build_metrics_json(
         snap_id=details['snap_id'],
         installed_base=installed_base,
-        metric_period=metric_requested['int'],
+        # Magic -1 otherwise for 7 days we get 8, for 30 days we get 31?
+        metric_period=metric_requested['int'] - 1,
```

Comment:
Was it wrong before? Is this the issue with API?

Reply:
We're asking for today - the period length. Which is inclusive, so we get 1 extra point. Rather then doing this magic number, I've added it to the logic file